### PR TITLE
ARCH-1619 - Cleanup tech debt & old patterns

### PR DESCRIPTION
**Pull request by danielle-casella-adams**
# Summary of PR changes

- Recompile using npm 8.15.0 and node v16.17.1
- Update actions/github package to latest version
- Convert octokit.list calls to use the paginate method instead of making multiple api calls in a loop
- Update the octokit await calls to use the await octokit.action.then().catch() pattern instead of a mix of that and try/catches.
- Update setFailed() calls to print the error.message rather than the error object

## PR Requirements

- [x] For JavaScript actions, the action has been recompiled.  
  - See the *Recompiling* section of the repository's README.md for more details.
  - This does not apply to Composite Run Steps actions unless a package.json is present and contains a build script.
- [x] For major, minor, or breaking changes, at least one of the commit messages contains the appropriate `+semver:` keywords.
  - See the *Incrementing the Version* section of the repository's README.md for more details.
- [x] The examples in the repository's `README.md` have been updated with the new version.  
  - See the *Incrementing the Version* section of the repository's README.md for more details on how the version will be incremented.
- [x] The action code does not contain sensitive information.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -154,27 +156,28 @@ async function getAllActiveItemsOnTheBoard() {
...
-    core.setFailed(`An error occurred retrieving the cards: ${error}`);
+    core.setFailed(`An error occurred retrieving the cards: ${error.message}`);
   }
   return [];
 }
 
 async function getActiveBranches() {
-  try {
-    const response = await octokit.rest.repos.listBranches({
+  await octokit
```
Comment (JosephDSchwartz) [inline src/main.js:165]: So I could be being super dumb right now, it is the morning after all, but I think we need this return statement added right? The old return statements got shoved into the anonymous function being passed to the `then` method.
```suggestion
  return await octokit
```

**Review by JosephDSchwartz — COMMENTED** [review]
I have the one question/potential suggestion, but otherwise I think this looks good!

File: src/main.js (hunk 1 above)
Comment (danielle-casella-adams) [reply src/main.js:165]: Yeah we need a fix for that.  Good catch!

Comment (danielle-casella-adams) [reply src/main.js:165]: This is the exact thing I was worried I would miss when I converted them all from try/catch to await/then/catch.

Comment (danielle-casella-adams) [reply src/main.js:166]: I made the change in a slightly different way but it's been pushed up.

**Review by JosephDSchwartz — APPROVED** [review]

File: src/main.js (hunk 1 above)
Comment (JosephDSchwartz) [reply src/main.js:166]: Your change matches the pattern of the other actions, so I think it makes more sense!

**Pull request merged**